---
title: OSError
permalink: /Python/OSError/
excerpt: "La OSError es una excepción que maneja errores del sistema operativo en Python."
date: 2025-01-23
last_modified_at: 2025-01-23
key: Python.O.OSError
category: Python
tags: [excepcion python,python 1]
image: "/assets/images/python.png"
sidebar:
  nav: python
---

## Descripción


La excepción `OSError` es una excepción base de [Python](https://www.manualweb.net/python/) que se produce cuando ocurre un error relacionado con operaciones del sistema operativo, como problemas al acceder a archivos, permisos denegados o recursos no disponibles. Esta excepción es la clase base para excepciones específicas del sistema operativo como [`FileNotFoundError`](https://www.w3api.com/Python/FileNotFoundError/), [`PermissionError`](https://www.w3api.com/Python/PermissionError/) y otros errores relacionados con E/S. `OSError` proporciona información detallada sobre el error específico del sistema operativo que causó la excepción.


Aunque `OSError` ha existido desde Python 1.5, la forma en que se maneja ha evolucionado:


### **Antes de Python 3.3**


`OSError` era una excepción general para manejar cualquier error del sistema operativo, como problemas de archivos, permisos, dispositivos, etc.


No existían excepciones más específicas. Todo error relacionado se capturaba con `OSError`.


### **Desde Python 3.3**


Se introdujeron excepciones más específicas que derivan de `OSError`, como: [`FileNotFoundError`](https://www.w3api.com/Python/FileNotFoundError/), [`PermissionError`](https://www.w3api.com/Python/PermissionError/) ,…


## Sintaxis


```python
exception OSError([arg])
```


## Ejemplo


Un ejemplo de cómo se puede producir y manejar un error `OSError` es intentar acceder a un archivo que no existe:


```python
try:
    file = open("archivo_inexistente.txt", "r")
except OSError as error:
    print(f"Se produjo un error: {error}")
    # Se produjo un error: [Errno 2] No such file or directory: 'archivo_inexistente.txt'
```


En este ejemplo, cuando intentamos abrir un archivo que no existe, [`Python`](https://www.manualweb.net/python/) lanza una `OSError` (específicamente [`FileNotFoundError`](https://www.w3api.com/Python/FileNotFoundError/) en Python 3.x). El bloque except captura la excepción y muestra el mensaje de error correspondiente.


## Artículos

